scania-digital-design-system/tegel
# feat(typography & logos): locally hosted files

## **Describe pull-request**  
The issue our users face is that we load font files via the CDN link set here in Europe, to be precise in Ireland.
That is a problem for users in China where they can not load files outside China.
We have placed the font files as assets in the mono repo and updated the stencil to copy them and make them available when shipping the code.
This way we increase the size of the package from 3.7 to 4.5MB but we make it more stable and independent. It is then really up to the application team to make sure they app is available to China or any other country. 
The same issue was with Logotypes used in the Header and Footer components.

**What is left**
- [ ] Document script usage for consumers how to copy over files

## **Issue Linking:**  
- **Jira:** Add ticket number after `CDEP-`: [CDEP-3530](https://tegel.atlassian.net/browse/CDEP-3530)    
- **Jira:** Add ticket number after `CDEP-`: [CDEP-3085](https://tegel.atlassian.net/browse/CDEP-3085)   

## **How to test**  
1. Check the preview link and see if Storybook loads fonts and logos currently. Use the network tab of the inspector to confirm font files are loaded and where they are coming from. 
2. Logos are located in Header and Footer component
3. Run this branch locally.
4. Do `npm run build` on `packages/core` level
5. Check `packages/core/dist/tegel/assets` to see if font files can be found.

## **Checklist before submission**

## **Suggested test steps**

## **Additional context**  
I noticed that a lot of tests have failed visually. It seems to me that Docker was sometimes loading different fonts when taking screenshots or it was taking them too early so not all screenshots were 100% the same as when you load the storybook locally. 
I have dropped all screenshots and created new ones to compare them here in PR so that others can test it and see if this branch gives a pass on unit tests. Let's see. 
Plan is to make beta release, send it over to person from support and ask them to try it out. 


[CDEP-3530]: https://tegel.atlassian.net/browse/CDEP-3530?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CDEP-3085]: https://tegel.atlassian.net/browse/CDEP-3085?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
